framework/src/auxkernels/TagVectorArrayVariableValueAux.C

```diff
@@ -0,0 +1,41 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TagVectorArrayVariableValueAux.h"
+
+registerMooseObject("MooseApp", TagVectorArrayVariableValueAux);
+
+InputParameters
+TagVectorArrayVariableValueAux::validParams()
+{
+  InputParameters params = TagAuxBase<ArrayAuxKernel>::validParams();
+
+  params.addRequiredParam<TagName>("vector_tag", "Tag Name this Aux works on");
+  params.addClassDescription("Couple a tagged vector, and return its array value.");
+  return params;
+}
+
+TagVectorArrayVariableValueAux::TagVectorArrayVariableValueAux(const InputParameters & parameters)
+  : TagAuxBase<ArrayAuxKernel>(parameters), _v(coupledVectorTagArrayValue("v", "vector_tag"))
+{
+  if (getArrayVar("v", 0)->feType() != _var.feType())
+    paramError("variable",
+               "The AuxVariable this AuxKernel is acting on has to have the same order and family "
+               "as the variable 'v'");
+  if (getArrayVar("v", 0)->count() != _var.count())
+    paramError("variable",
+               "The AuxVariable this AuxKernel is acting on has to have the same number of "
+               "components as the variable 'v'");
```

Comment:
this is a pretty common check in coupleable derived objects. Could you make a routine for it and share it with other coupledAux object?

should make these lines covered

Reply:
Defined a function `checkCoupledVariable` in `TagAuxBase` and reused in the derived objects.

Add the test cases to cover these checks.